app/src/test/java/com/hedvig/app/apollo/EmbarkStoryQueryParsingTest.kt

```diff
@@ -0,0 +1,191 @@
+@file:OptIn(ApolloExperimental::class)
+
+package com.hedvig.app.apollo
+
+import assertk.assertThat
+import assertk.assertions.isEqualTo
+import assertk.assertions.isNotNull
+import com.apollographql.apollo3.ApolloClient
+import com.apollographql.apollo3.annotations.ApolloExperimental
+import com.apollographql.apollo3.mockserver.MockServer
+import com.apollographql.apollo3.mockserver.enqueue
+import com.apollographql.apollo3.testing.runTest
+import com.hedvig.android.owldroid.graphql.EmbarkStoryQuery
+import com.hedvig.app.testdata.feature.embark.data.STANDARD_STORY
+import com.hedvig.app.testdata.feature.embark.data.STORY_WITH_GRAPHQL_MUTATION
+import com.hedvig.app.testdata.feature.embark.data.STORY_WITH_GRAPHQL_MUTATION_AND_SINGLE_VARIABLE
+import com.hedvig.app.testdata.feature.embark.data.STORY_WITH_GRAPHQL_QUERY_API_AND_GENERATED_VARIABLE
+import com.hedvig.app.testdata.feature.embark.data.STORY_WITH_GRAPHQL_QUERY_API_AND_SINGLE_VARIABLE
+import com.hedvig.app.testdata.feature.embark.data.STORY_WITH_NUMBER_ACTION
+import com.hedvig.app.testdata.feature.embark.data.STORY_WITH_PASSED_KEY_VALUE
+import com.hedvig.app.testdata.feature.embark.data.STORY_WITH_SELECT_ACTION_API_MULTIPLE_OPTIONS
+import com.hedvig.app.testdata.feature.embark.data.STORY_WITH_TEXT_ACTION_API
+import org.junit.Test
+
+class EmbarkStoryQueryParsingTest {
```

Comment:
Could we use the new generated test builders instead of constructing these ourselves? Would be so nice if we could use generated objects so that we dont have to change these when the schema changes

Reply:
Potentially yes, but there's a couple of concerns.

The way we're having builders for each type/fragment individually isn't something we can do with the test builders as I see it. The API they're providing is good for constructing a whole Operation.Data object in one go, so I think then it wouldn't map 1:1 to what we're doing now so it could make it more complicated.
With that said, if we use them as they're meant to be used, which is construct the entire object right there for the test you want to test, and not reuse them all over the place maybe it would work. But this is more work again.

As far as the API goes, it's also something that maybe isn't super convenient. Things aren't exactly type-safe when constructing the object as everything is a Map<String, Any?> and it's relying on providing you with typeSafe builders to not write whatever. For leaf items, the type is always just "String" and when you need to specify an object on a leaf item you need to manually do the deserialization from the item into the string representation it needs to be, like [here](https://github.com/HedvigInsurance/android/blob/86c04df55c8f5a21a41634698a6bbcabd9d5c9ad/app/src/test/java/com/hedvig/app/apollo/UpcomingAgreementQueryParsing.kt#L120-L122).
Plus in general the API is experimental and there's stuff that could change in a backwards incompatible way which may lead to even more migration work for us. And since we're using these generated models so heavily who knows if that would take days again.
Hence I decided to hold off such a big move and tried to make everything work the way they used to as much as I could.